### PR TITLE
Fixing version check for Openjdk

### LINKED ISSUE #87: Unable to get java version with OpenJDK
Hi.
I'm under archlinux and i can't run the resolver.

```
Failed to get Java version when running /usr/lib/jvm/default/bin/java
It is not be possible to verify your Java installation is new enough to compile with the latest Android SDK

Successfully executed '/usr/lib/jvm/default/bin/java -version'
stdout:

stderr:
openjdk version "1.8.0_144"
OpenJDK Runtime Environment (build 1.8.0_144-b01)
OpenJDK 64-Bit Server VM (build 25.144-b01, mixed mode)

exit code: 0

UnityEngine.Debug:LogWarning(Object)
Google.Logger:Log(String, LogLevel)
GooglePlayServices.PlayServicesResolver:Log(String, LogLevel)
GooglePlayServices.JavaUtilities:LogJdkVersionFailedWarning(String, String)
GooglePlayServices.JavaUtilities:CheckJdkForApiLevel()
GooglePlayServices.PlayServicesResolver:ResolveUnsafe(Action`1, Boolean)
GooglePlayServices.<Resolve>c__AnonStorey11:<>m__19()
GooglePlayServices.PlayServicesResolver:ExecuteNextResolveJob()
GooglePlayServices.PlayServicesResolver:Resolve(Action, Boolean, Action`1)
GooglePlayServices.PlayServicesResolver:ExecuteMenuResolve(Boolean)
GooglePlayServices.PlayServicesResolver:MenuResolve()
```

I can build android app with Android studio without problemes and build android unity project.

Is this a problem with openjdk ?

I'm using the last version : 1.2.59.0

### PR DESCRIPTION
**Pull request by skad**
- using multiline regex for parsing verion number
 - validate if java or openjdk

resolve #87

### CODE REVIEW

File: source/PlayServicesResolver/src/JavaUtilities.cs [hunk 1]
```diff
@@ -185,19 +186,12 @@ private static string FindJavaTool(string javaTool)
...
-                    }
-                    if (!float.TryParse(components[0] + "." + components[1], NumberStyles.Any,
-                                        CultureInfo.InvariantCulture, out majorMinorVersion)) {
-                        continue;
-                    }
-                }
+            string pattern = "^(?<model>java||openjdk) version \"(?<first>\\d).(?<major>\\d).(?<minor>\\d).*$";
+            
+            Match match = Regex.Match(result.message,pattern,RegexOptions.Multiline);
+            if(match.Success){
```
Comment (stewartmiles) [inline source/PlayServicesResolver/src/JavaUtilities.cs]: style nit: Space after "if" and before brace.

if (match.Success) {

File: source/PlayServicesResolver/src/JavaUtilities.cs [hunk 2]
```diff
@@ -185,19 +186,12 @@ private static string FindJavaTool(string javaTool)
...
-                    var components = versionString.Split(new [] { '.' });
-                    if (components.Length < 2) {
-                        continue;
-                    }
-                    if (!float.TryParse(components[0] + "." + components[1], NumberStyles.Any,
-                                        CultureInfo.InvariantCulture, out majorMinorVersion)) {
-                        continue;
-                    }
-                }
+            string pattern = "^(?<model>java||openjdk) version \"(?<first>\\d).(?<major>\\d).(?<minor>\\d).*$";
```
Comment (stewartmiles) [inline source/PlayServicesResolver/src/JavaUtilities.cs]: nit: I believe version is semver.  i.e major.minor.revision  Worth renaming the fields?

Comment (skad) [reply source/PlayServicesResolver/src/JavaUtilities.cs]: I wasn't sure, fixed it.

**Comment by stewartmiles** [comment]
One more minor thing.  Any chance you could squash this into a single commit?  Makes it easier to review the change in its' entirety.

```
git rebase -i
```

or if you upstream repo is the default "origin"

```
git rebase -i origin/master
```

There's a description here which is fair
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by skad** [comment]
done

**Comment by stewartmiles** [comment]
Great thanks for the patch.

**Pull request merged**